app/frontend/src/features/communities/events/CreateEventPage.tsx

```diff
@@ -0,0 +1,95 @@
+import { Error as GrpcError } from "grpc-web";
+import { Event } from "proto/events_pb";
+import { communityEventsBaseKey } from "queryKeys";
+import { useMutation, useQueryClient } from "react-query";
+import { service } from "service";
+import type { CreateEventInput } from "service/events";
+import dayjs from "utils/dayjs";
+import makeStyles from "utils/makeStyles";
+
+import { CREATE_EVENT, CREATE_EVENT_SUCCESS } from "./constants";
+import EventForm, { CreateEventData } from "./EventForm";
+
+const useStyles = makeStyles((theme) => ({}));
+
+export default function CreateEventPage() {
+  const queryClient = useQueryClient();
+  const {
+    mutate: createEvent,
+    error,
+    isLoading,
+    isSuccess,
+  } = useMutation<
+    Event.AsObject,
+    GrpcError,
+    CreateEventData,
+    { parentCommunityId?: number }
+  >(
+    (data) => {
+      let createEventInput: CreateEventInput;
+      const finalStartDate = data.startDate
+        .startOf("day")
+        .add(dayjs(data.startTime).get("hour"), "hour")
+        .add(dayjs(data.startTime).get("minute"), "minute")
+        .toDate();
+      const finalEndDate = data.endDate
+        .startOf("day")
+        .add(dayjs(data.endTime).get("hour"), "hour")
+        .add(dayjs(data.endTime).get("minute"), "minute")
+        .toDate();
+
+      if (data.isOnline) {
+        createEventInput = {
+          isOnline: data.isOnline,
+          title: data.title,
+          content: data.content,
+          photoKey: data.eventImage,
+          startTime: finalStartDate,
+          endTime: finalEndDate,
+          // TODO: not hardcode this and allow user to specify community ID?
+          parentCommunityId: 1,
+          link: data.link,
+        };
+      } else {
+        createEventInput = {
+          isOnline: data.isOnline,
+          title: data.title,
+          content: data.content,
+          photoKey: data.eventImage,
+          startTime: finalStartDate,
+          endTime: finalEndDate,
+          address: data.location.simplifiedName,
+          lat: data.location.location.lat,
+          lng: data.location.location.lng,
+        };
```

Comment:
missing parent community id?

Reply:
I think for offline events, the backend has some magic which figures out which community ID to assign the event to, hence it's not here

Reply:
Looking at the backend code, it's optional and if you don't include it, it automatically assigns a community as you said. But I think we definitely want to specify here - I thought something was broken because I was using the global community to create a physical event in New York, and then I couldn't see it. But I just found it in the NY community.

Reply:
How do we feel about the possibility of letting people create an event that is clearly not in the community though? E.g. they could theoretically create an offline event in Paris in the New York community

Reply:
In addition to the above question... doing this is would be a bit clunky at the moment under `/event/new`, since we don't know the community ID the user comes from if they go to the URL directly...

Should we not have the route and do the `#new` thing like we did for discussions instead?

Reply:
I think creating events outside the community is either fine or should be checked in the backend.

I would just do `/event/new/:communityId?` and if one isn't provided, do the automatic way.

Reply:
> if one isn't provided, do the automatic way.

As in, let the backend decide which community to put the event at as before?

Reply:
I've done it with route state, not put it under community wrapper though but I think that's ok if the event is being created for the community.